spec/locale.html

```diff
@@ -96,7 +100,7 @@ <h1>ApplyOptionsToTag ( _tag_, _options_ )</h1>
             1. Set _languageId_ to _languageId_ with the <emu-not-ref>substring</emu-not-ref> matched by the <code>unicode_script_subtag</code> Unicode locale nonterminal replaced by the string _script_.
         1. If _region_ is not *undefined*, then
           1. If _languageId_ does not contain a <emu-not-ref>substring</emu-not-ref> matched by the <code>unicode_region_subtag</code> Unicode locale nonterminal, then
-            1. Let _variants_ be the longest suffix of _languageId_ that is a consecutive sequence of <emu-not-ref>substrings</emu-not-ref> in which each element is a *"-"* followed by a <emu-not-ref>substring</emu-not-ref> that is matched by the <code>unicode_variant_subtag</code> Unicode locale nonterminal. If there is no such suffix, let _variants_ be the empty String.
+            1. Let _variants_ be the longest suffix of _languageId_ that is a consecutive sequence of <emu-not-ref>substrings</emu-not-ref> in which each element is a *"-"* followed by a <emu-not-ref>substring</emu-not-ref> that is matched by the <code>unicode_variant_subtag</code> Unicode locale nonterminal. If there is no such suffix, set _variants_ to the empty String.
```

Comment:
I've changed this to a `set`, because it's required by ecmarkup:

```
error: "variants" is already declared (re-declaration) at spec/locale.html:103:344:
  101 |         1. If _region_ is not *undefined*, then
  102 |           1. If _languageId_ does not contain a <emu-not-ref>substring</emu-not-ref> matched by the <code>unicode_region_subtag</code> Unicode locale nonterminal, then
> 103 |             1. Let _variants_ be the longest suffix of _languageId_ that is a consecutive sequence of <emu-not-ref>substrings</emu-not-ref> in which each element is a *"-"* followed by a <emu-not-ref>substring</emu-not-ref> that is matched by the <code>unicode_variant_subtag</code> Unicode locale nonterminal. If there is no such suffix, let _variants_ be the empty String.
      |                                                                                                                                                                                                                                                                                                                                                        ^
  104 |             1. Set _languageId_ to the string-concatenation of:
  105 |               * the longest <emu-not-ref>substring</emu-not-ref> of _languageId_ that is immediately followed by _variants_
  106 |               * *"-"*
```

Reply:
The wording is kind of weird anyway: "Do this thing. If you couldn't, do this other thing" seems kind of unnatural, to me. I would prefer to only give instructions which can be followed, rather than giving instructions which might be impossible to follow and then telling you how to deal with them. So maybe

```suggestion
            1. If there is a non-empty suffix of _languageId_ that is a consecutive sequence of <emu-not-ref>substrings</emu-not-ref> in which each element is a *"-"* followed by a <emu-not-ref>substring</emu-not-ref> that is matched by the <code>unicode_variant_subtag</code> Unicode locale nonterminal, let _variants_ be the longest such suffix; otherwise let _variants_ be the empty String.
```

Alternatively you could just make it clear that the longest suffix may in fact be the empty string:

```suggestion
            1. Let _variants_ be the (possibly empty) longest suffix of _languageId_ that is a consecutive sequence of <emu-not-ref>substrings</emu-not-ref> in which each element is a *"-"* followed by a <emu-not-ref>substring</emu-not-ref> that is matched by the <code>unicode_variant_subtag</code> Unicode locale nonterminal. 
```